### ISSUE TITLE
RFE : Permanent water level under the boat?

### ISSUE THREAD

**Issue by Gilletarom** [rgleason/weather_routing_pi#131]
This concerns WR users who
- use vector charts (O-Charts or other origin)
- have local tide data

We have in the settings of "Options/Charts/Vector-Charts", an adjustment concerning the height of safe water under the boat.
If we know the time of low tide and the time of high tide for a given place, it is possible to calculate the water level at a given time. This water level is always added to the depths indicated on the vector charts. So that actually changes the safe depth under a boat.

One could imagine that WR could use this modified depth to choose whether or not to let a road pass in a place on the map where the water level is low and less than the safety depth entered in the options/Charts.Vector -charts?

Gilletarom.

**Comment by Gilletarom** [comment]
Hello Dave, Rick,

I come back to my request #131. Perhaps, and even certainly, it is too complicated to implement and will require too much machine time in terms of the amount of calculation, given the gigantic number of probes present in the vector charts.

So, I only ask that, at each moment, we can know the height of water available under the boat and around the boat in a circle of fairly small diameter.

Would this be possible?

**Comment by rgleason** [comment]
Yes, it is only as good as the data in the chart of course.  I am not so sure this belongs in Weather-Routing, but of course the idea to use ENC chart water depth in lieu of the Detailed Background land perimeter is a good one and could possibly be implemented.

**Comment by sebastien-rosset** [comment]
This is somewhat related to https://github.com/OpenCPN/OpenCPN/issues/4387. Using vector charts to extract depth makes sense for a safety margin and I'll try to work on it at some point.

However, I'm skeptical about weather routing in shallow areas where the margin between the vessel's draft and the seabed is minimal, and depends on tidal predictions that have inherent uncertainties (Prediction Uncertainties, Charting Inaccuracies:, Temporal Precision, Operational Reality). Perhaps this would belong to route safety, i.e. identifying the risks of a given route.

In principle, we could create a hydrodynamic model for tidal current prediction, using a 3D computational mesh of the water body based on detailed bathymetry, use known tidal harmonics, incorporate river inflows where applicable, apply shallow water equations (2D) or Navier-Stokes equations (3D) or somehow ML to solve for water levels and current vectors throughout the domain. Then with that we might be able to incorporate routing.

Personally, I would love to have that capability to estimate tidal currents anywhere, but not for weather routing. Near the coastline, the local effects are much more important compared to the GRIB data.

**Comment by rgleason** [comment]
This is a great idea and at the moment a practical impossibility..   Therefore I am closing it.
There are also several duplicates.


"Would it be possible for WR to check the water depth and prevent the proposed route from passing over an area that is too shallow?"
